src/test/java/org/openrewrite/staticanalysis/TernaryOperatorsShouldNotBeNestedTest.java

```diff
@@ -0,0 +1,1315 @@
+        @Test
+        void doReplaceNestedOrTernaryInStreamWithIfInBlock() {
+            rewriteRun(
+              //language=java
+              java(
+                """
+                  import java.util.Set;
+                  import java.util.Arrays;
+                  import java.util.List;
+                  import java.util.stream.Collectors;
+                  class Test {
+                    public Set<String> makeASet() {
+                        List<String> s = Arrays.asList("a","b","c","nope");
+                        return s.stream().map(item -> item.startsWith("a") ? "a" : item.startsWith("b") ? "b" : "nope").collect(Collectors.toSet());
+                    }
+                  }
+                  """,
+                """
+                  import java.util.Set;
+                  import java.util.Arrays;
+                  import java.util.List;
+                  import java.util.stream.Collectors;
+                  class Test {
+                    public Set<String> makeASet() {
+                        List<String> s = Arrays.asList("a","b","c","nope");
+                        return s.stream().map(item -> {
+                            if (item.startsWith("a")) {
+                                return "a";
+                            }
+                            return item.startsWith("b") ? "b" : "nope";
+                        }).collect(Collectors.toSet());
+                    }
+                  }
+                  """
+              )
+            );
+        }
+
+        @Test
+        void doReplaceNestedOrTernaryInStreamContainingComments() {
+            rewriteRun(
+              //language=java
+              java(
+                """
+                  import java.util.Set;
+                  import java.util.Arrays;
+                  import java.util.List;
+                  import java.util.stream.Collectors;
+                  class Test {
+                    public Set<String> makeASet() {
+                        List<String> s = Arrays.asList("a","b","c","nope");
+                        return s.stream().map( /* look a lambda */ item ->
+                             //look a ternary
+                             item.startsWith("a") ? "a" : item.startsWith("b") ? "b" : "nope"
+                             ).collect(Collectors.toSet());
+                    }
+                  }
+                  """,
+                """
+                  import java.util.Set;
+                  import java.util.Arrays;
+                  import java.util.List;
+                  import java.util.stream.Collectors;
+                  class Test {
+                    public Set<String> makeASet() {
+                        List<String> s = Arrays.asList("a","b","c","nope");
+                        return s.stream().map( /* look a lambda */ item ->
+                                //look a ternary
+                                {
+                                    if (item.startsWith("a")) {
```

Comment:
Intuitively I'd have expected this comment on the `if`, not on the `block` Would that be hard to change to reduce friction on any PRs this generates?

Reply:
I remember playing with this and doubting where to put it. It's not hard to move and I think I agree with you. It looks better on the if.